katharsis-core/src/test/java/io/katharsis/utils/PropertyUtilsTest.java

```diff
@@ -192,10 +194,31 @@ public void onListValueForSetPropertyShouldGetConverted() throws Exception {
     	Bean bean = new Bean();
 
         // WHEN
-    	PropertyUtils.setProperty(bean, "setProperty", Arrays.asList("1", "2"));
-
-        // THEN
-        assertThat(bean.getSetProperty().size()).isEqualTo(2);
+        PropertyUtils.setProperty(bean, "setProperty", Arrays.asList("4", "1", "3", "2"));
+
+        // THEN
+        assertThat(bean.getSetProperty().size()).isEqualTo(4);
+
+        // confirm the order has been persevered
+        Iterator<String> setPropertyIterator = bean.getSetProperty().iterator();
+        for (int i = 0; i < bean.getSetProperty().size(); i++) {
```

Comment:
A better test would be to create a LinkedHashSet with the expected values and just assert equals on both sets